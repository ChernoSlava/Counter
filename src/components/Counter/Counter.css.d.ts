export const Counter: string;
export const Counter_zero: string;
export const Counter_lesszero: string;
export const Counter_abovezero: string;
