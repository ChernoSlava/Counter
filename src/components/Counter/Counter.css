.Counter {
    font-size: 24px;
    color: black;
    font-weight: bold;
}

.Counter_zero {

    color: black;

}

.Counter_lesszero {
    color: red;
}

.Counter_abovezero {
    color: green;
}

