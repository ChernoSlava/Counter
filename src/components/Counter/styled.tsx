import styled from 'styled-components';
import { ThemeType } from '@theme';

export type CounterType = 'zero' | 'lesszero' | 'abovezero';

export const CounterStyled = styled.div<{
  type: CounterType;
}>`
  font-size: 24px;
  color: black;
  font-weight: bold;
  ${props =>
    props.type === 'abovezero' &&
    `
        color: ${(props.theme as ThemeType).colors.green10};
    `}
  ${props =>
    props.type === 'lesszero' &&
    `
        color: ${(props.theme as ThemeType).colors.red10};
    `}
    ${props =>
    props.type === 'zero' &&
    `
        color: ${(props.theme as ThemeType).colors.black};
    `}
`;
