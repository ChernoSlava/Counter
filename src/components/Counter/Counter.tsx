import React from "react";
import styles from './Counter.css';
import { CounterStyled, CounterType,  } from "./styled";

export const Counter: React.FC<{
    title: string;
    type: CounterType;
}>  = ({ title, type }): JSX.Element => {
    return (
        <CounterStyled type={type}>{title}</CounterStyled>
    )
}
