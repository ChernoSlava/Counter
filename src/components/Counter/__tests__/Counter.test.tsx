/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import { render } from '@testing-library/react';

import React from 'react';
import { Counter, CounterProps } from '../Counter';

const getUI: React.FC<CounterProps> = (props?: CounterProps): JSX.Element => {
  return <Counter title="0" type="zero" {...(props || {})} />;
};

describe('Counter tests', () => {
  it('should render correct counter type zero', () => {
    const { container } = render(
      getUI({
        title: '0',
        type: 'zero',
      }),
    );

    expect(container).toMatchSnapshot();
  });
  it('should render correct counter type lesszero', () => {
    const { container } = render(
      getUI({
        title: '-2',
        type: 'lesszero',
      }),
    );

    expect(container).toMatchSnapshot();
  });
  it('should render correct counter type abovezero', () => {
    const { container } = render(
      getUI({
        title: '5',
        type: 'abovezero',
      }),
    );

    expect(container).toMatchSnapshot();
  });
});
