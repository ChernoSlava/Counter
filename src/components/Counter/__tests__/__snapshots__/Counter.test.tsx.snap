// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter tests should render correct counter type abovezero 1`] = `
.c0 {
  font-size: 24px;
  color: black;
  font-weight: bold;
  color: green;
}

<div>
  <div
    class="c0"
    type="abovezero"
  >
    5
  </div>
</div>
`;

exports[`Counter tests should render correct counter type lesszero 1`] = `
.c0 {
  font-size: 24px;
  color: black;
  font-weight: bold;
  color: indianred;
}

<div>
  <div
    class="c0"
    type="lesszero"
  >
    -2
  </div>
</div>
`;

exports[`Counter tests should render correct counter type zero 1`] = `
.c0 {
  font-size: 24px;
  color: black;
  font-weight: bold;
  color: black;
}

<div>
  <div
    class="c0"
    type="zero"
  >
    0
  </div>
</div>
`;
